doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
 Zephyr build system, as described in more detail in other sections on
 this page.
 
-Zephyr depends on several categories of modules, including:
+To be classified as a candidate for being included in the default list of
+modules, an external project is required to have its own life-cycle outside
+the Zephyr Project, that is, reside in its own repository, and have its own
+contribution and maintainance workfow and release process. Zephyr modules
+should not contain code that is written exclusively for Zephyr.
+
+  .. note::
+
+     Code which is written for Zephyr and works exclusively with Zephyr
+     should not be submitted as a Zephyr module; instead, it should be
+     part of the main Zephyr tree.
+
+Modules to be included in the default manifest of the Zephyr project need to
+provide functionality or features endorsed and approved by the project technical
+steering committee and should comply with the
+:ref:`module licensing requirements<modules_licensing>` and
+:ref:`contribute guidelines<modules_contributing>`. They should also have a
+Zephyr developer that is commited to maintain the module code base.
+
+Zephyr depends on several categories of modules, including but not limited to:
 
 - Debugger integration
 - Silicon vendor Hardware Abstraction Layers (HALs)
 - Cryptography libraries
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+This page summarizes a list of policies and best practices which aim at
+better organizing the workflow in Zephyr modules.
+
+Module Repositories
+*******************
+
+* All modules included in the default manifest shall be hosted in repositories
+  under the Zephyr Project Github organization.
+
+* Module repository names should follow the convention of using lowercase
```

Comment:
This is a convention we have followed so far. We do link to the upstream repo directly in the description of each of the Zephyr project repos.

Reply:
> This is a convention we have followed so far.

If we are grandfathering in exceptions, then clearly not.

> We do link to the upstream repo directly in the description of each of the Zephyr project repos.

So now I have to click around inside of https://github.com/zephyrproject-rtos instead of just copy/pasting the upstream repository name?

Reply:
This was discussed in the previous meeting.
@mbolivar-nordic we will change the policy as follows
- module repositories that are tracking existing external project repositories should have the exact name as upstream
- module repositories created in zephyr organization that are not directly tracking an external repository, e.g. tracking multiple ones, or tracking none (in case the ext project is maintained outside github) shall follow the proposed naming conventions.
- we don't retro-update existing module repositories.

Thanks. I will make this change.

Reply:
applied @mbolivar-nordic 

```
Module repository names should follow the convention of using lowercase
  letters and dashes instead of underscores. This rule will apply to all
  new module repositories, except for repositories that are directly
  tracking external projects hosted in Github; such module repositories
  may be named as their external project counterparts.
```

Reply:
Thanks @ioannisg